zebra/zebra_vxlan.c

```diff
@@ -3427,6 +3427,19 @@ static void zvni_mac_del_hash_entry(struct hash_bucket *bucket, void *arg)
 			zvni_mac_uninstall(wctx->zvni, mac);
 
 		zvni_mac_del(wctx->zvni, mac);
+	} else if ((wctx->flags & DEL_LOCAL_MAC)
```

Comment:
Suggestion: IMO, It would be good if we can make it more readable.
Lacks readability with more and more conditions added on single ifelse. 
Shall we write it a bit clear way writing a basic utility to know if it can be deleted.

bool needs_deletion;
needs_deletion = zvni_mac_needs_deletion_in_walk_ctx(wctx, mac);

if (needs_deletion) {
  		if (wctx->upd_client && (mac->flags & ZEBRA_MAC_LOCAL)) {
			zvni_mac_send_del_to_client(wctx->zvni->vni,
						    &mac->macaddr);
		}
		if (wctx->uninstall)
			zvni_mac_uninstall(wctx->zvni, mac);

		zvni_mac_del(wctx->zvni, mac);
}

---------

static inline bool zvni_mac_needs_deletion_in_walk_ctx(wctx, mac) {
 if (cond1)
   return TRUE;
if (cond2)
   return TRUE;
if (cond3)
/* justification for condition as a comment if needed/special case */
-
-
-
return FALSE;
}